### PR TITLE
Add parRCB

### PR DESCRIPTION
**Pull request by thilinarmtb**

### CODE REVIEW

File: 3rd_party/parRSB/install [hunk 1]
```diff
@@ -12,14 +12,11 @@ if [ -f ./lib/libparRSB.a ]; then
   exit 0
 fi
 
-if [ ! -f v$VER.tar.gz ]; then
-  wget --no-check-certificate -O v$VER.tar.gz https://github.com/Nek5000/parRSB/archive/v$VER.tar.gz
+if [ ! -d parRSB ]; then
```
Comment (thilinarmtb) [inline 3rd_party/parRSB/install]: Undo this change.

File: core/makenek.inc [hunk 2]
```diff
@@ -315,6 +315,7 @@ do
      PARRSB=1
      PARRSB_OPT=" DESTDIR=.. UNDERSCORE=$UNDERSCORE MPI=$MPI" 
      PARRSB_OPT+=" GSLIBPATH=$SOURCE_ROOT_GSLIB"
+     USR_LFLAGS+=" -Wl,-rpath,${SOURCE_ROOT_PARRSB}/lib"
```
Comment (thilinarmtb) [inline core/makenek.inc]: Undo this change.

File: 3rd_party/parRSB/install [hunk 3]
```diff
@@ -12,14 +12,11 @@ if [ -f ./lib/libparRSB.a ]; then
...
+  git clone https://github.com/thilinarmtb/parRSB.git -b parcon
 fi
 
-rm -rf parRSB 2>/dev/null
-mkdir parRSB 
-tar -zxvf v$VER.tar.gz -C ./parRSB --strip-components=1
```
Comment (thilinarmtb) [inline 3rd_party/parRSB/install:21]: Undo these changes.

File: 3rd_party/gslib/install [hunk 4]
```diff
@@ -28,5 +28,5 @@ mkdir gslib
 tar -zxvf v$VER.tar.gz -C ./gslib --strip-components=1
 cd gslib
 set -x
-make -j4 $GSLIB_OPT
+CFLAGS="-O2 -fPIC" make -j4 $GSLIB_OPT
```
Comment (stgeke) [inline 3rd_party/gslib/install]: Use CFLAGS of PARRSB set in makenek.inc

File: core/INPUT [hunk 5]
```diff
@@ -49,7 +49,7 @@ c
      $               ,ifcyclic,ifmoab,ifcoup, ifvcoup, ifusermv,ifreguo
      $               ,ifxyo_,ifaziv,ifneknek,ifneknekm,ifneknekc
      $               ,ifcvfld(ldimt1),ifdp0dt
-     $               ,ifmpiio,ifrich,ifvvisp
+     $               ,ifmpiio,ifrich,ifvvisp,ifparrsb,ifparrcb
```
Comment (stgeke) [inline core/INPUT]: Use integer meshParitioner instead of ifparrsb etc.

File: core/PARDICT [hunk 6]
```diff
@@ -106,3 +106,4 @@ c
      &  pardictkey(99) / 'GENERAL:MAXNUMHISTORYPOINTS' /
      &  pardictkey(100)/ 'GENERAL:FILTERMODES' /
      &  pardictkey(101)/ 'PRESSURE:SOLVER' /
+     &  pardictkey(102)/ 'GENERAL:PARTITIONER' /
```
Comment (stgeke) [inline core/PARDICT]: Move in MESH section

File: core/map2.f [hunk 7]
```diff
@@ -160,18 +160,36 @@ subroutine get_vert_map(vertex,nlv,wk,mdw,ndw)
 
       common /nekmpi/ mid,mp,nekcomm,nekgroup,nekreal
 
-      logical ifparrsb
+      integer ipartmode
```
Comment (stgeke) [inline core/map2.f]: use meshParitioner (see comment above(

File: core/partitioner.c [hunk 8]
```diff
@@ -333,6 +247,149 @@ void printPartStat(long long *vtx, int nel, int nv, comm_ext ce)
...
+  rsb=mode & 1;
+  rcb=mode & 2;
+
+  opt[0] = 1;
+  opt[1] = 2; /* verbosity */
+  opt[2] = 0;
+
+  if(rcb){
+    if(comm.id==0)
+      printf("Before parRCB:\n");
```
Comment (stgeke) [inline core/partitioner.c]: I don't think we need this kind of print statements

File: core/reader_par.f [hunk 9]
```diff
@@ -145,6 +145,8 @@ subroutine setDefaultParam
          idpss(i) = -1
       enddo 
 
+      meshPartitioner=1 ! RSB
```
Comment (stgeke) [inline core/reader_par.f]: RCB+RSB should be the default

File: core/reader_par.f [hunk 10]
```diff
@@ -822,6 +824,16 @@ subroutine par_read(ierr)
...
       enddo
 
+c set partitioner options
+      call finiparser_getString(c_out,'mesh:partitioner',ifnd)
+      call capit(c_out,132)
+      if(index(c_out,'RSB').eq.1) then
+         meshPartitioner=1
+      else if(index(c_out,'RCB').eq.1) then
+         meshPartitioner=2
+      else if (index(c_out,'HYBRID').eq.1) then
```
Comment (stgeke) [inline core/reader_par.f:834]: How about parMetis?

**Comment by stgeke** [comment]
Replaced by #674
